### PR TITLE
taocpp-pegtl: conan v2 support

### PR DESCRIPTION
**Pull request by rbrich**
Specify library name and version:  **taocpp-pegtl**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/taocpp-pegtl/3.x.x/conanfile.py [hunk 1]
```diff
@@ -51,28 +50,27 @@ def lazy_lt_semver(v1, v2):
 
         minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
         if not minimum_version:
-            self.output.warn("{} {} requires C++17. Your compiler is unknown. Assuming it supports C++17.".format(self.name, self.version))
+            self.output.warn(f"{self.ref} requires C++17. Your compiler is unknown. Assuming it supports C++17.")
```
Comment (SpaceIm) [inline recipes/taocpp-pegtl/3.x.x/conanfile.py:53]: self.output.warn doesn't exist in conan v2, it's self.output.warning, but I advice to drop this warning, it's useless.

File: recipes/taocpp-pegtl/3.x.x/conanfile.py [hunk 2]
```diff
@@ -51,28 +50,27 @@ def lazy_lt_semver(v1, v2):
...
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++17, which your compiler does not support.")
 
-        compiler_version = tools.Version(self.settings.compiler.version)
+        compiler_version = Version(self.settings.compiler.version)
         if self.version == "3.0.0" and self.settings.compiler == "clang" and \
            compiler_version >= "10" and compiler_version < "12":
-            raise ConanInvalidConfiguration("{} {} doesn't support filesystem experimental".format(self.name, self.version))
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support filesystem experimental")
 
         if self.options.boost_filesystem and (self.options["boost"].header_only or self.options["boost"].without_filesystem):
```
Comment (SpaceIm) [inline recipes/taocpp-pegtl/3.x.x/conanfile.py:62]:
```suggestion
        if self.options.boost_filesystem and (self.dependencies["boost"].options.header_only or self.dependencies["boost"].options.without_filesystem):
```

File: recipes/taocpp-pegtl/3.x.x/test_v1_package/test_package.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2014-2020 Dr. Colin Hirsch and Daniel Frey
```
Comment (SpaceIm) [inline recipes/taocpp-pegtl/3.x.x/test_v1_package/test_package.cpp:1]: this duplication of code can be avoided, see package templates in docs

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Do not mind me... I an just testing something about how the GitHub API track approvals

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**